### PR TITLE
Diagnose -i arg

### PR DESCRIPTION
**Pull request by sjavora**
Addresses [this FIXME](https://github.com/apple/swift-driver/blob/1e1c30534eb6af42bf66763c5ed6d918e2f56a60/Sources/SwiftDriver/Driver/Driver.swift#L691). 

I created a new test for this as the assert didn't seem to fit into `testPrimaryOutputKinds`, but I'll of course change it if desired.

I'm not exactly sure if `DriverKind.usage` should be used here, but the values seem to correspond to [this code](https://github.com/apple/swift/blob/d930599394b27fcac77a90b63e23372c6b31cd7a/lib/Driver/Driver.cpp#L95) in the original driver.

### CODE REVIEW

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -707,6 +706,17 @@ extension Driver {
   }
 }
 
+extension Diagnostic.Message {
+  public static func error_i_mode(_ driverKind: DriverKind) -> Diagnostic.Message {
+    .error(
+      """
+      the flag '-i' is no longer required and has been removed;
```
Comment (harlanhaskins) [inline Sources/SwiftDriver/Driver/Driver.swift]: Can you add a `\` at the end of this line, so it doesn't introduce a newline into the diagnostic text?
```suggestion
      the flag '-i' is no longer required and has been removed; \
```

Comment (sjavora) [reply Sources/SwiftDriver/Driver/Driver.swift]: Of course, I thought that's what the original code was doing, but my C++ knowledge is fairly limited.

**Comment by harlanhaskins** [comment]
@swift-ci please test

**Comment by harlanhaskins** [comment]
Thanks @sjavora!

**Pull request merged**